velox/common/file/File.cpp

```diff
@@ -75,20 +75,28 @@ LocalReadFile::LocalReadFile(std::string_view path) {
       "open failure in LocalReadFile constructor, {} {} {}.",
       fd_,
       path,
-      strerror(errno));
+      folly::errnoStr(errno));
   const off_t rc = lseek(fd_, 0, SEEK_END);
   VELOX_CHECK_GE(
       rc,
       0,
       "fseek failure in LocalReadFile constructor, {} {} {}.",
       rc,
       path,
-      strerror(errno));
+      folly::errnoStr(errno));
   size_ = rc;
 }
 
 LocalReadFile::LocalReadFile(int32_t fd) : fd_(fd) {}
 
+LocalReadFile::~LocalReadFile() {
+  const int ret = close(fd_);
+  if (ret < 0) {
+    LOG(WARNING) << "close failure in LocalReadFile destructor: " << ret << ", "
+                 << folly::errnoStr(errno);
+  }
+}
+
```

Comment:
Does it mean that before this change the handles would only be closed after the process is gone?